### DISCUSSION TITLE
How do I enforce separate input layout for windows client machine?

### DISCUSSION THREAD

**Discussion by TheDreamsWind** [debauchee/barrier#1862] [Q&A]
For context: I have a mac server with windows client with barrier v2.4.0 installed on both machines. Both machines have multiple input layouts and in fact I can even successfully switch between layouts with keyboard shortcuts on both machines, however the client machine always uses the server strokes, i.e. even if i switch from non-english to english on the windows (client) machine, while macOS server has non-english layout (or vice-versa), the server relays it's input to the client (no matter which layout is currently selected), so In order to change input layout on the client machine I have to switch to server, switch input layout, and then switch back to client:
```
macOS server layout #1 + win client layout #1 => layout #1 input on the client
macOS server layout #2 + win client layout #2 => layout #2 input on the client
macOS server layout #1 + win client layout #2 => layout #1 input on the client
macOS server layout #2 + win client layout #1 => layout #2 input on the client
```
Previously, when I had the same macOS server but linux client, the client was able to interpret input in accordance with its own selected input layout (but only if the macOS server had english layout selected):
```
layout #1 = "en-US qwerty" input layout
macOS server layout #1 + linux client layout #1 => layout #1 input on the client
macOS server layout #2 + linux client layout #2 => layout #2 input on the client
macOS server layout #1 + linux client layout #2 => layout #2 input on the client
macOS server layout #2 + linux client layout #1 => layout #2 input on the client
```

Is there a way to enable the same logic for the windows client?

**Comment by Kxchrtn** [comment]
> For context: I have a mac server with windows client with barrier v2.4.0 installed on both machines. Both machines have multiple input layouts and in fact I can even successfully switch between layouts with keyboard shortcuts on both machines, however the client machine always uses the server strokes, i.e. even if i switch from non-english to english on the windows (client) machine, while macOS server has non-english layout (or vice-versa), the server relays it's input to the client (no matter which layout is currently selected), so In order to change input layout on the client machine I have to switch to server, switch input layout, and then switch back to client:
> 
> ```
> macOS server layout #1 + win client layout #1 => layout #1 input on the client
> macOS server layout #2 + win client layout #2 => layout #2 input on the client
> macOS server layout #1 + win client layout #2 => layout #1 input on the client
> macOS server layout #2 + win client layout #1 => layout #2 input on the client
> ```
> 
> Previously, when I had the same macOS server but linux client, the client was able to interpret input in accordance with its own selected input layout (but only if the macOS server had english layout selected):
> 
> ```
> layout #1 = "en-US qwerty" input layout
> macOS server layout #1 + linux client layout #1 => layout #1 input on the client
> macOS server layout #2 + linux client layout #2 => layout #2 input on the client
> macOS server layout #1 + linux client layout #2 => layout #2 input on the client
> macOS server layout #2 + linux client layout #1 => layout #2 input on the client
> ```
> 
> Is there a way to enable the same logic for the windows client?






Hello, sorry to bother you. Can you let me know please, did you find any solution of this problem? Thanks!

**Reply by TheDreamsWind** [reply]
I gave up on this. It seems to me that the project is currently abandoned
